### DISCUSSION TITLE
How can I create a Popup that takes whole bottom of the screen?

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [xamarin/XamarinCommunityToolkit#992] [Q&A] [answered]
Something like this:

![image](https://user-images.githubusercontent.com/6609929/109237600-8a316180-77da-11eb-90cc-c182c7b45e86.png)

**Comment by Cfun1** [comment, marked as answer]
Create a custom size that takes the whole screen width and for example takes 40% of screen Height:
```
public static Size Custom => new Size((DeviceDisplay.MainDisplayInfo.Width / DeviceDisplay.MainDisplayInfo.Density), 
                                0.4 * (DeviceDisplay.MainDisplayInfo.Height / DeviceDisplay.MainDisplayInfo.Density));
```
Look at https://github.com/xamarin/XamarinCommunityToolkit/blob/develop/samples/XCT.Sample/Pages/Views/Popups/PopupSize.cs use your custom size in Size property like in sample app: https://github.com/xamarin/XamarinCommunityToolkit/blob/e976b79fc29f9f9458f9e656b774cb3e03cdf254/samples/XCT.Sample/Pages/Views/Popups/SimplePopup.xaml#L6 and finally set the position to bottom center also like shown in the sample app https://github.com/xamarin/XamarinCommunityToolkit/blob/e976b79fc29f9f9458f9e656b774cb3e03cdf254/samples/XCT.Sample/Pages/Views/Popups/PopupPositionPage.xaml#L103 or directly set the vertical and horizontal alignment like it done in the command here https://github.com/xamarin/XamarinCommunityToolkit/blob/e976b79fc29f9f9458f9e656b774cb3e03cdf254/samples/XCT.Sample/ViewModels/Views/Popups/PopupPositionViewModel.cs#L53-L56

I know I am kinda of linked you to the sample code, hope globally it is helpful.

**Reply by maxkoshevoi** [reply]
Thank you, it worked. I was trying to achieve it be "AndExpand" in `HorizontalOptions`. What does "Expand" even do then?

**Reply by maxkoshevoi** [reply]
Also, how can I make height of the popup autosize based on content height? Can I control how it appears somehow (speed and animation)? I want it to pop from the bottom.

**Reply by maxkoshevoi** [reply]
> Thank you, it worked.

No, wait, when page is rotated, size is not recalculated:
![image](https://user-images.githubusercontent.com/6609929/109296625-4caff100-7839-11eb-9135-451411d9de16.png)
![image](https://user-images.githubusercontent.com/6609929/109296641-533e6880-7839-11eb-97f6-80e529b91899.png)

**Reply by Cfun1** [reply]
> > Thank you, it worked.
> 
> No, wait, when page is rotated, size is not recalculated:
> ![image](https://user-images.githubusercontent.com/6609929/109296625-4caff100-7839-11eb-9135-451411d9de16.png)
> ![image](https://user-images.githubusercontent.com/6609929/109296641-533e6880-7839-11eb-97f6-80e529b91899.png)
> 
> 

I believe you ca override OnSize allocated check screen orientation using [essatials](https://docs.microsoft.com/en-us/xamarin/essentials/device-display?tabs=android#main-display-info) do some magic accordingly:

```
protected override void OnSizeAllocated(double width, double height) { 
  var mainDisplayInfo = DeviceDisplay.MainDisplayInfo;
  var orientation = mainDisplayInfo.Orientation;
  If (orientation == Xamarin.Essentials.DisplayOrientation.Landscape)
  {
    popup.Size = new Size((DeviceDisplay.MainDisplayInfo.Height / DeviceDisplay.MainDisplayInfo.Density), 
                                0.4 * (DeviceDisplay.MainDisplayInfo.Width / DeviceDisplay.MainDisplayInfo.Density));
    }
    else
   {
  popup.Size = new Size((DeviceDisplay.MainDisplayInfo.Width / DeviceDisplay.MainDisplayInfo.Density), 
                                0.4 * (DeviceDisplay.MainDisplayInfo.Height / DeviceDisplay.MainDisplayInfo.Density));
         }
```
